sda-download/api/sda/sda.go

```diff
@@ -207,6 +207,11 @@ func Files(c *gin.Context) {
 
 // Download serves file contents as bytes
 func Download(c *gin.Context) {
+	if c.Param("type") != "encrypted" && !config.Config.App.AllowUnencryptedDownload {
+		c.String(http.StatusForbidden, "request to unencrypted file not allowed")
+
+		return
+	}
```

Comment:
```suggestion
	if c.GetHeader("Client-Public-Key") == "" && !config.Config.App.AllowUnencryptedDownload {
		c.String(http.StatusBadRequest, "no public key in request")

		return
	}
```

Target the header instead, it will make this more future proof.

Reply:
I don't know exactly how we will refactor in the future. Will we remove `c.Param("type")` ?
For the moment, I think the condition `c.Param("type") != "encrypted"` is not suitable since the user can easily bypass the restriction of `config.Config.App.AllowUnencryptedDownload` by providing `Client-Public-Key` in the header when download unencrypted file, e.g. 

```
curl -H "Client-Public-Key: $pubkey" -H "Authorization: Bearer $token" -k http://localhost:8443/s3/DATASET0001//htsnexus_test_NA12878.bam --output t2.bam 
```

Reply:
This would require some more refactoring, though, right? The public-key header is only checked for the `s3-encrypted` endpoint.
```sh
$ curl -o /dev/null -s -w "%{http_code}\n" -H "Authorization: Bearer $token" -H -k http://localhost:8443/s3/DATASET0001//htsnexus_test_NA12878.bam 
403
$ curl -o test.bam -s -w "%{http_code}\n" -H "Authorization: Bearer $token" -H -k http://localhost:8443/s3/DATASET0001//htsnexus_test_NA12878.bam  -H 'Client-Public-Key: abc' # test.bam is downloaded, unencrypted
200 
```

Reply:
As Malin @MalinAhlberg pointed out as well, I think for the moment it is better to use the condition `c.Param("type") != "encrypted"` instead of checking `Client-Public-Key` in the header, and leave that for the future refactoring. I will change the return code from `http.StatusForbidden` to `http.StatusBadRequest`. Does it makes sense @jbygdell ?